src/test/scala/uk/gov/nationalarchives/dp/client/DataProcessorTest.scala

```diff
@@ -0,0 +1,216 @@
+package uk.gov.nationalarchives.dp.client
+
+import cats.MonadError
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers._
+
+abstract class DataProcessorTest[F[_]](implicit cme: MonadError[F, Throwable]) extends AnyFlatSpec {
+  def valueFromF[T](value: F[T]): T
+
+  "fragmentUrls" should "return fragments" in {
+    val input =
+      <MetadataResponse>
+        <AdditionalInformation>
+          <Metadata>
+            <Fragment>test1</Fragment>
+            <Fragment>test2</Fragment>
+          </Metadata>
+        </AdditionalInformation>
+      </MetadataResponse>
+    val response = new DataProcessor[F]().fragmentUrls(input)
+    val values = valueFromF(response)
+
+    values.size should equal(2)
+    values.head should equal("test1")
+    values.last should equal("test2")
+  }
+
+  "fragmentUrls" should "return an empty list when there are no fragments" in {
+    val input =
+      <MetadataResponse>
+        <AdditionalInformation>
+          <Metadata>
+          </Metadata>
+        </AdditionalInformation>
+      </MetadataResponse>
+    val response = new DataProcessor[F]().fragmentUrls(input)
+    val values = valueFromF(response)
+
+    values.size should equal(0)
+  }
+
+  "fragments" should "return multiple fragments" in {
+    def fragment(i: Int) = <Fragment>Fragment
+      {i}
+    </Fragment>
+
+    def fragmentContainer(i: Int) =
+      <MetadataResponse>
+        <MetadataContainer>
+          <Content>
+            {fragment(i)}
+          </Content>
+        </MetadataContainer>
+      </MetadataResponse>
+
+    val fragmentsF =
+      new DataProcessor[F]().fragments(Seq(fragmentContainer(1), fragmentContainer(2)))
+    val fragments = valueFromF(fragmentsF)
+
+    fragments.size should equal(2)
+    fragments.head.trim should equal(fragment(1).toString)
+    fragments.last.trim should equal(fragment(2).toString)
+  }
+
+  "fragments" should "return an empty list if there is no content" in {
```

Comment:
Is this possible? And if so, should this be an error?

Reply:
It's not I don't think, thanks. This is the metadata endpoint which means there must be a metadata ref which means there must be metadata. I've changed it to throw an error if there's an empty list and updated the tests.